arch/x86/slide_hash_sse2.c

```diff
@@ -15,18 +15,36 @@
 
 static inline void slide_hash_chain(Pos *table, uint32_t entries, const __m128i wsize) {
     table += entries;
-    table -= 8;
+    table -= 16;
+    __m128i value0, value1, result0, result1;
 
-    do {
-        __m128i value, result;
+    /* ZALLOC allocates this pointer unless the user chose a custom allocator.
+     * Our alloc function is aligned to 64 byte boundaries */
+    if (UNLIKELY(((uintptr_t)table & 15) != 0)) {
```

Comment:
@KungFuJesus am I correct in assuming that if #1096 gets merged, then we don't need to account for custom allocators, and we wouldn't then need this branch?

Reply:
In theory, yes.  We could then cut the size of the function in half, basically.  This assumes that there's no weird corner cases that could end up messing with it.

Reply:
At present, anyway, slide_hash is only ever called on members allocated with the ZALLOC macro, so, I could remove the if entirely, sub in a possible sanity check at the beginning of the function (or least just a debug level assertion), and say this commit depends on #1096 ?

I see LONGEST_MATCH contains some Assert() function, should I be using that?  Does this disappear while in debug mode?  I don't expect a user to encounter this issue as these things are only ever allocated by us, so a debug level assert is probably appropriate here.

Reply:
With regard to shortening the length of the compiled code: to be honest I'm a little surprised that the compiler, after it does this inlining, isn't smart enough to identify this loop is just repeated.  It would be trivial to swap a pointer and a length variable and jump to the label containing the loop.  Maybe I should try that to see if there's any discernible improvement.  At the very least it'd reduce L1 cache pollution.

Reply:
Yeah, no appreciable difference to runtime, just significantly smaller code size.